fms/modules/attention.py

```diff
@@ -136,7 +136,10 @@ def forward(
         if k is None and v is None:
             qkv = q
         elif k is not None and v is not None:
-            qkv = torch.cat((q, k, v), dim=-1)
+            # Note: for encoder/decoder models with cross attn, this line may need to be changed as kv will be fused and
+            # q will be kept separate. For now, we are not running any encoder/decoder models through this layer and
```

Comment:
I had been testing out some models locally that relied on this functionality (and fixed a related bug I encountered while doing so)... this is a BC-breaking change.
If there's no way to make it work for both scenarios, we may want to add a distinct fused module